### PR TITLE
Latency logging for every stage of pipeline

### PR DESCRIPTION
**Pull request by fzhar**

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
when input is single JPEG image decoding time is nan

File: demos/common/cpp/pipelines/src/async_pipeline.cpp [hunk 1]
```diff
@@ -67,15 +67,17 @@ int64_t AsyncPipeline::submitData(const InputData& inputData, const std::shared_
...
 
     request->SetCompletionCallback(
-        [this, frameID, request, internalModelData, metaData] {
+        [this, frameID, request, internalModelData, metaData, startTime] {
             request->SetCompletionCallback([]{});
 
             {
                 std::lock_guard<std::mutex> lock(mtx);
-
+                this->inferenceMetrics.update(startTime);
```
Comment (eizamaliev) [inline demos/common/cpp/pipelines/src/async_pipeline.cpp:80]:
```suggestion
                inferenceMetrics.update(startTime);
```
I'd prefer using `this` everywhere, but ... :)

File: demos/object_detection_demo/cpp/main.cpp [hunk 2]
```diff
@@ -424,6 +430,17 @@ int main(int argc, char *argv[]) {
         //// --------------------------- Report metrics -------------------------------------------------------
         slog::info << slog::endl << "Metric reports:" << slog::endl;
         metrics.printTotal();
+        slog::info << slog::endl << "Avg time:\n";
```
Comment (eizamaliev) [inline demos/object_detection_demo/cpp/main.cpp:433]:
```suggestion
        slog::info << slog::endl << "Average time:\n";
```

**Review by eizamaliev — CHANGES_REQUESTED** [review]

**Comment by eizamaliev** [comment]
Would you add renderer metrics in other Pipeline\* demos in this PR?

**Comment by fzhar** [comment]
> Would you add renderer metrics in other Pipeline* demos in this PR?

As we've agreed before, these measurements will be implemented for object_detection_demo only in this PR, other demos will be modified in separate PRs

**Review by Wovchena — APPROVED** [review]

**Review by vladimir-dudnik — APPROVED** [review]

**Review by eizamaliev — APPROVED** [review]

**Pull request merged**